File: Chalkable.Web/app/chlk/models/district/District.js
REQUIRE('chlk.models.common.ChlkDate');
REQUIRE('chlk.models.id.DistrictId');
REQUIRE('chlk.models.id.BgTaskId');
NAMESPACE('chlk.models.district', function () {
    "use strict";

    /** @class chlk.models.district.District*/
    CLASS(
        'District', [
            chlk.models.id.DistrictId, 'id',
            String, 'name',
            [ria.serialize.SerializeProperty('sisurl')],
            String, 'sisUrl',
            [ria.serialize.SerializeProperty('dbname')],
            String, 'dbName',
            [ria.serialize.SerializeProperty('sisusername')],
            String, 'sisUserName',
            [ria.serialize.SerializeProperty('sispassword')],
            String, 'sisPassword',
            [ria.serialize.SerializeProperty('sissystemtype')],
            Number, 'sisSystemType',

            [ria.serialize.SerializeProperty('failcounter')],
            Number, 'failCounter',

            [ria.serialize.SerializeProperty('syncfrequency')],
            Number, 'syncFrequency',

            [ria.serialize.SerializeProperty('maxsyncfrequency')],
            Number, 'maxSyncFrequency',

            [ria.serialize.SerializeProperty('failDelta')],
            Number, 'failDelta',

            [ria.serialize.SerializeProperty('processingid')],
            chlk.models.id.BgTaskId, 'processingId',

            [ria.serialize.SerializeProperty('processingcreated')],
            chlk.models.common.ChlkDate, 'processingCreated',

            [ria.serialize.SerializeProperty('processingscheduled')],
            chlk.models.common.ChlkDate, 'processingScheduled',

            [ria.serialize.SerializeProperty('processingstarted')],
            chlk.models.common.ChlkDate, 'processingStarted',


            [ria.serialize.SerializeProperty('completedid')],
            chlk.models.id.BgTaskId, 'completedId',

            [ria.serialize.SerializeProperty('completedcreated')],
            chlk.models.common.ChlkDate, 'completedCreated',

            [ria.serialize.SerializeProperty('completedscheduled')],
            chlk.models.common.ChlkDate, 'completedScheduled',

            [ria.serialize.SerializeProperty('completedstarted')],
            chlk.models.common.ChlkDate, 'completedStarted',

            [ria.serialize.SerializeProperty('completedcompleted')],
            chlk.models.common.ChlkDate, 'completedCompleted',


            [ria.serialize.SerializeProperty('failedid')],
            chlk.models.id.BgTaskId, 'failedId',

            [ria.serialize.SerializeProperty('failedcreated')],
            chlk.models.common.ChlkDate, 'failedCreated',

            [ria.serialize.SerializeProperty('failedscheduled')],
            chlk.models.common.ChlkDate, 'failedScheduled',

            [ria.serialize.SerializeProperty('failedstarted')],
            chlk.models.common.ChlkDate, 'failedStarted',

            [ria.serialize.SerializeProperty('failedcompleted')],
            chlk.models.common.ChlkDate, 'failedCompleted',


            [ria.serialize.SerializeProperty('canceledid')],
            chlk.models.id.BgTaskId, 'canceledId',

            [ria.serialize.SerializeProperty('canceledcreated')],
            chlk.models.common.ChlkDate, 'canceledCreated',

            [ria.serialize.SerializeProperty('canceledscheduled')],
            chlk.models.common.ChlkDate, 'canceledScheduled',

            [ria.serialize.SerializeProperty('canceledstarted')],
            chlk.models.common.ChlkDate, 'canceledStarted',

            [ria.serialize.SerializeProperty('canceledcompleted')],
            chlk.models.common.ChlkDate, 'canceledCompleted',

            [ria.serialize.SerializeProperty('newid')],
            chlk.models.id.BgTaskId, 'newId',

            [ria.serialize.SerializeProperty('newcreated')],
            chlk.models.common.ChlkDate, 'newCreated',

            [ria.serialize.SerializeProperty('newscheduled')],
            chlk.models.common.ChlkDate, 'newScheduled'

        ]);
});

File: Chalkable.Data.Master/Model/District.cs
﻿using System;
using Chalkable.Data.Common;

namespace Chalkable.Data.Master.Model
{
    public class District
    {
        public const string ID_FIELD = "Id";
        public const string SERVER_URL_FIELD = "ServerUrl";
        public const string SIS_URL_FIELD = "SisUrl";
        public const string SIS_DISTRICT_IF_FIELD = "SisDistrictId";
        public const string LAST_SYNC_FIELD = "LastSync";

        [PrimaryKeyFieldAttr]
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string SisUrl { get; set; }
        public string SisUserName { get; set; }
        public string SisPassword { get; set; }
        public int Status { get; set; }
        public string TimeZone { get; set; }
        public string ServerUrl { get; set; }
        public string SisRedirectUrl { get; set; }
        public DateTime? LastSync { get; set; }
        public int MaxSyncTime { get; set; }
        public int SyncLogFlushSize { get; set; }
        public int SyncHistoryDays { get; set; }
        
        [NotDbFieldAttr]
        public bool IsDemoDistrict { get; set; }
    }

    public enum DistrictStatus
    {
        Created = 0,
        DataImported = 1,
        InvitedUser = 2,
        TeacherLogged = 3,
        InvitedStudent = 4,
        StudentLogged = 5,
        PayingCustomer = 6
    }

    public class DistrictSyncStatus : District
    {
        public Guid? ProcessingId { get; set; }
		public DateTime? ProcessingCreated { get; set; }
		public DateTime? ProcessingStarted { get; set; }
		public Guid? CompletedId { get; set; }
		public DateTime? CompletedCreated { get; set; }
		public DateTime? CompletedStarted { get; set; }
		public DateTime? CompletedCompleted { get; set; }
		public Guid? FailedId { get; set; }
		public DateTime? FailedCreated { get; set; }
		public DateTime? FailedStarted { get; set; }
        public DateTime? FailedCompleted { get; set; }
        public Guid? CanceledId { get; set; }
        public DateTime? CanceledCreated { get; set; }
        public DateTime? CanceledStarted { get; set; }
        public DateTime? CanceledCompleted { get; set; }
    }
}